File: src/main/java/spring/webmvc/presentation/exception/InsufficientQuantityException.java
package spring.webmvc.presentation.exception;

import org.springframework.http.HttpStatus;

public class InsufficientQuantityException extends AbstractHttpException {

	public InsufficientQuantityException(String productName, int requestedQuantity, int availableStock) {
		super(
			"%s 상품의 재고가 부족합니다. (요청: %d, 재고: %d)".formatted(productName, requestedQuantity, availableStock),
			HttpStatus.CONFLICT
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/exception/InsufficientQuantityException.kt
package spring.web.kotlin.presentation.exception

import org.springframework.http.HttpStatus

class InsufficientQuantityException(productName: String, requestedQuantity: Int, availableStock: Int) :
    AbstractHttpException("$productName 상품의 재고가 부족합니다. (요청: $requestedQuantity, 재고: $availableStock)", HttpStatus.CONFLICT)